spec/02-integration/09-hybrid_mode/01-sync_spec.lua

```diff
@@ -288,6 +288,76 @@ for _, strategy in helpers.each_strategy() do
 
           proxy_client:close()
         end)
+
+        it('does not sync plugins when service with plugin-enabled route is disabled', function()
+          local admin_client = helpers.admin_client(10000)
+          finally(function()
+            admin_client:close()
+          end)
+
+          -- create service
+          local res = assert(admin_client:post("/services", {
+            body = { name = "mockbin-service2", url = "https://127.0.0.1:15556/request", },
+            headers = {["Content-Type"] = "application/json"}
+          }))
+          local body = assert.res_status(201, res)
+          local json = cjson.decode(body)
+          local service_id = json.id
+
+          -- create route
+          res = assert(admin_client:post("/services/mockbin-service2/routes", {
+            body = { paths = { "/soon-to-be-disabled" }, },
+            headers = {["Content-Type"] = "application/json"}
+          }))
+          local body = assert.res_status(201, res)
+          local json = cjson.decode(body)
+
+          local route_id = json.id
+
+          -- add a plugin for route
+          res = assert(admin_client:post("/routes/" .. route_id .. "/plugins", {
+            body = { name = "bot-detection" },
+            headers = {["Content-Type"] = "application/json"}
+          }))
+          assert.res_status(201, res)
+
+          -- test route
+          helpers.wait_until(function()
+            local proxy_client = helpers.http_client("127.0.0.1", 9002)
+
+            res = proxy_client:send({
+              method  = "GET",
+              path    = "/soon-to-be-disabled",
+            })
+
+            local status = res and res.status
+            proxy_client:close()
+            if status == 200 then
+              return true
+            end
+          end, 10)
+
+          -- disable service
+          local res = assert(admin_client:patch("/services/" .. service_id, {
+            body = { enabled = false, },
+            headers = {["Content-Type"] = "application/json"}
+          }))
+          assert.res_status(200, res)
+          -- as this is testing a negative behavior, there is no sure way to wait
+          -- this can probably be optimizted
+          ngx.sleep(2)
+
+          local proxy_client = helpers.http_client("127.0.0.1", 9002)
+
+          -- test route again
+          res = assert(proxy_client:send({
+            method  = "GET",
+            path    = "/soon-to-be-disabled",
+          }))
```

Comment:
Instead of `ngx.sleep`, you can wrap this code in a function using `helpers.wait_until()`, returning `status == 404`. When `status` is actually `404`, `wait_until` will finish. If `status` never is `404`, `wait_until` will timeout.

I don't remember now if the timeout is enough to fail the test, but you can look for examples, `wait_until` is widely used in the integration and plugins tests.